packages/zoe/src/zoe.js

```diff
@@ -150,7 +150,9 @@ import { makeTables } from './state';
  * `{ waived: null }`
  * `{ onDemand: null }`
  * `{ afterDeadline: { timer :Timer<Deadline>, deadline :Deadline } }
+ * @type {Object} Timer
```

Comment:
I think these need to be `@typedef`s, and moved to a different place so that they don't interfere with the definition of the current object's properties.

If it works for you, I'm really surprised, and my understanding is incomplete!

Reply:
It looks like you're right. I think this improved things, but then in https://github.com/Agoric/agoric-sdk/pull/1041, which is layered on top of this, I fixed it better. 

It turns out that that fix is still not quite sufficient. coveredCall tries to destructure `exit.afterDeadline.deadline`, and doesn't understand the conjunction  I ended up using. But we should talk about that there.

I'm going to abandon this PR.